File: reverse.py
def is_ascending_digits(num):
    num_str = str(num)
    if len(num_str) <= 1:
        return False  # Числа из одной цифры не считаются

    for i in range(1, len(num_str)):
        if num_str[i] <= num_str[i - 1]:
            return False
    return True


def reverse_number(num):
    num_str = str(num)
    reversed_str = num_str[::-1]  # Более Pythonic способ переворота строки
    try:
        return int(reversed_str)
    except ValueError:
        print(f"Превышение диапазона int для числа {num}", file=sys.stderr)
        return 0  # Или другое значение по умолчанию


import sys

input_str = input("Введите последовательность чисел, разделенных пробелами: ")
numbers_str = input_str.split()
numbers = []
for num_str in numbers_str:
    try:
        numbers.append(int(num_str))
    except ValueError:
        print(f"Некорректный ввод: {num_str} не является числом.", file=sys.stderr)

result = []
for num in numbers:
    if is_ascending_digits(num):
        result.append(reverse_number(num))

print("Результат: ", end="")
if not result:
    print("В последовательности нет чисел, соответствующих условию.")
else:
    print(*result)  # Более Pythonic способ вывода списка с пробелами

File: reverse.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func isAscendingDigits(num int) bool {
	numStr := strconv.Itoa(num)
	if len(numStr) <= 1 {
		return false // Числа из одной цифры не считаются
	}

	for i := 1; i < len(numStr); i++ {
		if numStr[i] <= numStr[i-1] {
			return false
		}
	}
	return true
}

func reverseNumber(num int) int {
	numStr := strconv.Itoa(num)
	reversedStr := ""
	for _, r := range numStr {
		reversedStr = string(r) + reversedStr
	}

	reversedNum, err := strconv.Atoi(reversedStr)
	if err != nil {
		fmt.Fprintln(os.Stderr, "Превышение диапазона int для числа", num)
		return 0 // Или другое значение по умолчанию
	}
	return reversedNum
}

func main() {
	reader := bufio.NewReader(os.Stdin)
	fmt.Print("Введите последовательность чисел, разделенных пробелами: ")
	input, _ := reader.ReadString('\n')
	input = strings.TrimSpace(input)

	numbersStr := strings.Split(input, " ")
	numbers := make([]int, 0)

	for _, numStr := range numbersStr {
		num, err := strconv.Atoi(numStr)
		if err != nil {
			fmt.Fprintln(os.Stderr, "Некорректный ввод:", numStr, "не является числом.")
			continue // Пропускаем некорректный ввод
		}
		numbers = append(numbers, num)
	}

	result := make([]int, 0)

	for _, num := range numbers {
		if isAscendingDigits(num) {
			result = append(result, reverseNumber(num))
		}
	}

	fmt.Print("Результат: ")
	if len(result) == 0 {
		fmt.Println("В последовательности нет чисел, соответствующих условию.")
	} else {
		for i, num := range result {
			fmt.Print(num)
			// HERE is the change
			if i < len(result)-1 {
				fmt.Print(" ")
			}
		}
		fmt.Println()
	}
}
